Frameworks/UIKit/UIImage.mm

```diff
@@ -714,102 +710,108 @@ - (void)drawAsPatternInRect:(CGRect)pos {
-                                        getImage(img)->Backing()->Height() - img->_imageInsets.top,
-                                        img->_imageInsets.right,
-                                        img->_imageInsets.top),
-                               makeRect(ii->size.size.width - img->_imageInsets.right,
-                                        ii->size.size.height - img->_imageInsets.top,
-                                        img->_imageInsets.right,
-                                        img->_imageInsets.top));
-    }
 
-    // Coordinates flipped on Y
-    if (getImage(img)->Backing()->Height() - img->_imageInsets.top - img->_imageInsets.bottom > 0) {
-        CGContextDrawImageRect(context,
-                               getImage(img),
-                               makeRect(0,
-                                        img->_imageInsets.bottom,
-                                        img->_imageInsets.left,
-                                        getImage(img)->Backing()->Height() - img->_imageInsets.top - img->_imageInsets.bottom),
-                               makeRect(0,
-                                        img->_imageInsets.bottom,
-                                        img->_imageInsets.left,
-                                        ii->size.size.height - img->_imageInsets.top - img->_imageInsets.bottom));
-        if (getImage(img)->Backing()->Width() - img->_imageInsets.left - img->_imageInsets.right > 0) {
+        if (src->size.width - leftCap - rightCap > 0) {
+            // MidHeightMidWidth
             CGContextDrawImageRect(context,
-                                   getImage(img),
-                                   makeRect(img->_imageInsets.left,
-                                            img->_imageInsets.bottom,
-                                            getImage(img)->Backing()->Width() - img->_imageInsets.left - img->_imageInsets.right,
-                                            getImage(img)->Backing()->Height() - img->_imageInsets.top - img->_imageInsets.bottom),
-                                   makeRect(img->_imageInsets.left,
-                                            img->_imageInsets.bottom,
-                                            ii->size.size.width - img->_imageInsets.left - img->_imageInsets.right,
-                                            ii->size.size.height - img->_imageInsets.top - img->_imageInsets.bottom));
+                                   cgImg,
+                                   makeRect(leftCap, botCap, (src->size.width - leftCap - rightCap), (src->size.height - topCap - botCap)),
+                                   makeRect((dst->origin.x + leftCap),
+                                            (dst->origin.y + botCap),
+                                            (dst->size.width - leftCap - rightCap),
+                                            (dst->size.height - topCap - botCap)));
         } else {
             assert(0);
```

Comment:
> assert(0); [](start = 11, length = 11)

although this is old code, it would be much better to Throw and exception with some information on why this is an issue.
It would provide much more information rather than just asserting.